Add generator plugin support along with support for alacritty generator

sekhat opened the pull request:
This adds a very simple "plugin" architecture for specifying additional generators.

Allows users to configure the default plugin/generator to use, while also allowing them to plugin their own custom generators.

And adds very simple support for generating an alacritty colors section

sekhat committed:
Add alacritty generator

sekhat committed:
Update README

sekhat committed:
Add simple plugin architecture to allow adding extra terminal generators

psliwka commented on lua/termcolors/plugins/alacritty.lua:
```diff
@@ -0,0 +1,44 @@
+    table.insert(config, "    blue: " .. quote_str(colorscheme["color12"]))
+    table.insert(config, "    mageneta: " .. quote_str(colorscheme["color13"]))
+    table.insert(config, "    cyan: " .. quote_str(colorscheme["color14"]))
+    table.insert(config, "    white: " .. quote_str(colorscheme["color15"]))
```
`scrape_current_colorscheme` is not guaranteed to return all 16 base terminal colors. In fact, my current colorscheme of choice (https://github.com/NTBBloodbath/doom-one.nvim) defines only `color0`..`color14`, causing this code to crash with `attempt to concatenate local 'str' (a nil value)` when trying to read `color15`. Some colorschemes may not define terminal colors at all. We need to handle this somehow.

sekhat replied:
Ok, so rather than solve the difficult problem. 

For now I just make sure not to write out entries for missing colours to avoid throwing an error.

psliwka commented on lua/termcolors.lua:
```diff
@@ -77,27 +79,32 @@ function termcolors.scrape_current_colorscheme()
-	return config
-end
+function termcolors.show(plugin_name)
+    if plugin_name == nil or plugin_name == "" then
```
You seem to have mixed space-based indentation into this tab-indented file. Could you please make it tab-indented everywhere (including newly introduced files)?

sekhat replied:
Done

psliwka reviewed (changes requested):
I love it, thanks! I can see one major issue though, and one minor styling nit. Please see my comments :)

sekhat committed:
Fix mixed indentation

sekhat committed:
Stop outputting missing terminal color entries

Not all colorschemes correctly set all 16 terminal color values, so an
error occurs in the alacritty generator if they are missing.

This is a quick fix to stop the current crash by ignoring generating output for
missing colors in the colorscheme.

psliwka reviewed (approved):
Thanks again!
